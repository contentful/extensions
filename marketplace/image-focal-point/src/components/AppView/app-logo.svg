<svg xmlns="http://www.w3.org/2000/svg" width="30" height="30" viewBox="0 0 30 30">
  <defs>
    <linearGradient id="a" x1="50%" x2="50%" y1="100%" y2="0%">
      <stop offset="0%" stop-color="#3C80CF"/>
      <stop offset="100%" stop-color="#5B9FEF"/>
    </linearGradient>
  </defs>
  <g fill="none" fill-rule="evenodd">
    <rect width="30" height="30" fill="url(#a)" rx="3"/>
    <g transform="translate(3 3)">
      <g stroke="#FFFFFF" stroke-linecap="round" stroke-linejoin="round" stroke-width="2" transform="translate(0 2)">
        <circle cx="12" cy="10" r="4.5"/>
        <polyline points="23.5 3.5 23.5 .5 20.5 .5"/>
        <polyline points="23.5 16.5 23.5 19.5 20.5 19.5"/>
        <polyline points=".5 3.5 .5 .5 3.5 .5"/>
        <polyline points=".5 16.5 .5 19.5 3.5 19.5"/>
      </g>
      <rect width="24" height="24"/>
    </g>
  </g>
</svg>
