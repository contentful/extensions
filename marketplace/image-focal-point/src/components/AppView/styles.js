import tokens from '@contentful/forma-36-tokens';
import { css } from 'emotion';

export const styles = {
  body: css({
    height: 'auto',
    minHeight: '65vh',
    margin: '0 auto',
    marginTop: tokens.spacingXl,
    padding: `${tokens.spacingXl} ${tokens.spacing2Xl}`,
    maxWidth: tokens.contentWidthText,
    backgroundColor: tokens.colorWhite,
    zIndex: '2',
    boxShadow: '0px 0px 20px rgba(0, 0, 0, 0.1)',
    borderRadius: '2px'
  }),
  background: css({
    display: 'block',
    position: 'absolute',
    zIndex: '-1',
    top: '0',
    width: '100%',
    height: '300px',
    backgroundColor: '#599ded' // corresponds to logo
  }),
  featuresListItem: css({
    listStyleType: 'disc',
    marginLeft: tokens.spacingM
  }),
  light: css({
    color: tokens.colorTextLight,
    marginTop: tokens.spacingM
  }),
  logo: css({
    width: '100%',
    display: 'flex',
    justifyContent: 'center',
    marginTop: tokens.spacingXl,
    marginBottom: tokens.spacingXl,
    '& > img': css({
      width: '80px'
    })
  }),
  paragraph: css({
    marginTop: tokens.spacingM
  }),
  heading: css({
    marginBottom: tokens.spacingM
  }),
  input: css({
    marginTop: tokens.spacingM
  }),
  list: css({
    marginTop: tokens.spacingXs
  })
};
