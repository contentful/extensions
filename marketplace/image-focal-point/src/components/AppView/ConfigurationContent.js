import React from 'react';
import { List, ListItem, Heading, Paragraph } from '@contentful/forma-36-react-components';
import { Divider } from '../Divider';
import { styles } from './styles';

export const ConfigurationContent = () => (
  <>
    <Heading className={styles.heading}>Getting started</Heading>
    <Paragraph>Follow the next steps to create a new image entry with focal point data:</Paragraph>
    <List className={styles.list}>
      <ListItem>Go to the &quot;Content&quot; page</ListItem>
      <ListItem>
        Create a new entry of type &quot;Image with Focal Point&quot; (or the name you chose during
        the installation)
      </ListItem>
      <ListItem>Fill in the required fields and publish it</ListItem>
    </List>
    <Paragraph className={styles.paragraph}>
      Follow the next steps to reference the newly created image with focal point entry in another
      content type:
    </Paragraph>
    <List className={styles.list}>
      <ListItem>Go to the &quot;Content model&quot; page</ListItem>
      <ListItem>
        Edit the content type that needs to reference the image with the focal point
      </ListItem>
      <ListItem>Create a new field of type &quot;Reference&quot;</ListItem>
      <ListItem>
        Set a validation rule requiring that the content type the reference points to is of type
        &quot;Image with Focal Point&quot;
      </ListItem>
      <ListItem>
        Optionally set the appearance of the new reference field to &quot;Entry Card&quot; in its
        settings
      </ListItem>
      <ListItem>Go back to the &quot;Content&quot; page</ListItem>
      <ListItem>
        Create/edit an entry of the content type referencing the image with the focal point
      </ListItem>
      <ListItem>
        Set the reference field you added to point to the image with focal point entry
      </ListItem>
    </List>
    <Divider />
    <Heading className={styles.heading}>Uninstallation and cleanup</Heading>
    <Paragraph>
      If you uninstall the Image Focal Point app you will have to manually clean up the content type
      we created for you during the installation. To do that follow the next steps:
    </Paragraph>
    <List className={styles.list}>
      <ListItem>Go to the &quot;Content&quot; page</ListItem>
      <ListItem>
        Delete any entries of type &quot;Image with Focal Point&quot; (or the name you chose during
        installation)
      </ListItem>
      <ListItem>Go to the &quot;Content model&quot; page</ListItem>
      <ListItem>
        Edit the content type the app created, select &quot;Actions&quot; from the top menu and
        &quot;Delete&quot;
      </ListItem>
    </List>
  </>
);
