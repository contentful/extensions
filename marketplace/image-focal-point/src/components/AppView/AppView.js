import React, { Component } from 'react';
import PropTypes from 'prop-types';
import camelCase from 'lodash.camelcase';
import { Heading, Typography, Paragraph } from '@contentful/forma-36-react-components';

import { ConfigurationContent } from './ConfigurationContent';
import { InstallationContent } from './InstallationContent';
import { Divider } from '../Divider';
import { styles } from './styles';

import appLogo from './app-logo.svg';

const makeContentType = (contentTypeId, contentTypeName) => ({
  sys: {
    id: contentTypeId
  },
  name: contentTypeName,
  displayField: 'title',
  fields: [
    {
      id: 'title',
      name: 'Title',
      required: true,
      type: 'Symbol'
    },
    {
      id: 'image',
      name: 'Image',
      required: true,
      type: 'Link',
      linkType: 'Asset'
    },
    {
      id: 'focalPoint',
      name: 'Focal point',
      required: true,
      type: 'Object'
    }
  ]
});

export class AppView extends Component {
  static propTypes = {
    sdk: PropTypes.object.isRequired
  };

  state = {
    isInstalled: false,
    allContentTypesIds: [],
    contentTypeId: camelCase('Image with Focal Point'),
    contentTypeName: 'Image with Focal Point',
    isContentTypeIdPristine: true
  };

  async componentDidMount() {
    const { space, app } = this.props.sdk;

    const [isInstalled, allContentTypes] = await Promise.all([
      app.isInstalled(),
      space.getContentTypes()
    ]);

    const allContentTypesIds = allContentTypes.items.map(({ sys: { id } }) => id);

    // Following eslint error is caused due to using async/await
    // eslint-disable-next-line react/no-did-mount-set-state
    this.setState({ isInstalled, allContentTypesIds }, () => app.setReady());

    app.onConfigure(this.installApp);
    app.onConfigurationCompleted(err => {
      if (!err) {
        this.setState({ isInstalled: true });
      }
    });
  }

  installApp = async () => {
    const { sdk } = this.props;
    const { isInstalled, allContentTypesIds, contentTypeId, contentTypeName } = this.state;

    if (isInstalled) {
      sdk.notifier.success('The app is already fully configured.');
      return false;
    }

    if (!contentTypeName) {
      sdk.notifier.error('Provide a name for the content type.');
      return false;
    }

    const isContentTypeIdTaken = allContentTypesIds.includes(contentTypeId);
    if (isContentTypeIdTaken) {
      sdk.notifier.error(
        `ID "${contentTypeId}" is taken. Try a different name for the content type`
      );
      return false;
    }

    let contentType = null;
    try {
      const data = makeContentType(contentTypeId, contentTypeName);
      contentType = await sdk.space.createContentType(data);
    } catch (error) {
      sdk.notifier.error(`Failed to create content type "${contentTypeName}"`);
      return false;
    }

    // Set the newly created content type's state to "Published"
    try {
      await sdk.space.updateContentType(contentType);
    } catch (error) {
      sdk.notifier.error(`Failed to publish content type "${contentTypeName}"`);
      return false;
    }

    return {
      targetState: {
        EditorInterface: {
          [contentType.sys.id]: {
            controls: [{ fieldId: 'focalPoint' }]
          }
        }
      }
    };
  };

  onContentTypeNameChange = ({ target: { value } }) =>
    this.setState(oldState => ({
      ...(oldState.isContentTypeIdPristine && { contentTypeId: camelCase(value) }),
      contentTypeName: value
    }));

  onContentTypeIdChange = ({ target: { value } }) =>
    this.setState({
      isContentTypeIdPristine: false,
      contentTypeId: value
    });

  render() {
    const { isInstalled, allContentTypesIds, contentTypeId, contentTypeName } = this.state;

    return (
      <>
        <div className={styles.background} />
        <div className={styles.body}>
          <Typography>
            <Heading className={styles.heading}>About Image Focal Point</Heading>
            <Paragraph>
              The Image Focal Point app allows you to associate focal point data with uploaded image
              assets, to achieve better cropping amongst different devices and screen sizes.
            </Paragraph>
            <Divider />
            {isInstalled && <ConfigurationContent />}
            {!isInstalled && (
              <InstallationContent
                allContentTypesIds={allContentTypesIds}
                contentTypeId={contentTypeId}
                contentTypeName={contentTypeName}
                onContentTypeNameChange={this.onContentTypeNameChange}
                onContentTypeIdChange={this.onContentTypeIdChange}
              />
            )}
          </Typography>
        </div>
        <div className={styles.logo}>
          <img src={appLogo} alt="logo" />
        </div>
      </>
    );
  }
}
