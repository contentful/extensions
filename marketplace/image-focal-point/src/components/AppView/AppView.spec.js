import React from 'react';
import { cleanup, render, wait, configure, fireEvent } from '@testing-library/react';

import mockProps from '../../test/mockProps';
import { AppView } from './AppView';

configure({ testIdAttribute: 'data-test-id' });

const props = {
  sdk: {
    ...mockProps.sdk,
    space: {
      getContentTypes: jest.fn(() => ({
        items: []
      }))
    },
    platformAlpha: {
      app: {
        onConfigure: jest.fn(),
        isInstalled: jest.fn(),
        setReady: jest.fn()
      }
    }
  }
};

describe('AppView', () => {
  afterEach(() => cleanup());

  it('should render the loading app view', () => {
    const { container } = render(<AppView {...props} />);
    expect(container).toMatchSnapshot();
  });

  describe('when the app is not installed', () => {
    beforeEach(() =>
      props.sdk.platformAlpha.app.isInstalled.mockImplementation(() => Promise.resolve(false))
    );

    it('should render the app view with installation screen', async () => {
      const appView = render(<AppView {...props} />);
      await wait();
      expect(appView.container).toMatchSnapshot();
    });

    it('should render inline validation if the content type id is taken', async () => {
      props.sdk.space.getContentTypes.mockImplementation(() => ({
        items: [{ sys: { id: 'imageWithFocalPoint' } }]
      }));
      const { getByTestId } = render(<AppView {...props} />);
      await wait();
      expect(getByTestId('content-type-name')).toMatchSnapshot();
      expect(getByTestId('content-type-id')).toMatchSnapshot();
    });

    it('should update the content type id field if it is pristine and the content type name is changed', async () => {
      const { getByTestId } = render(<AppView {...props} />);
      await wait();
      fireEvent.change(getByTestId('content-type-name-input'), { target: { value: 'Test Name' } });
      expect(getByTestId('content-type-id-input').value).toEqual('testName');
    });

    it('should not update the content type id field if it is dirty and the content type name is changed', async () => {
      const { getByTestId } = render(<AppView {...props} />);
      await wait();
      fireEvent.change(getByTestId('content-type-id-input'), { target: { value: 'someTestId' } });
      fireEvent.change(getByTestId('content-type-name-input'), { target: { value: 'Test Name' } });
      expect(getByTestId('content-type-id-input').value).toEqual('someTestId');
    });
  });

  describe('when the app is installed', () => {
    it('should render the app view with configuration screen', async () => {
      props.sdk.platformAlpha.app.isInstalled.mockImplementation(() => true);
      const appView = render(<AppView {...props} />);
      await wait();
      expect(appView.container).toMatchSnapshot();
    });
  });
});
