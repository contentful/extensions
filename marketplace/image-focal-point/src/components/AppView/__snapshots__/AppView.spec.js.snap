// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppView should render the loading app view 1`] = `
<div>
  <div
    class="css-1cyywdw"
  />
  <div
    class="css-jwwzlh"
  >
    <div
      class="Typography__Typography___1ZUfE"
      data-test-id="cf-ui-text-container"
    >
      <h1
        class="Heading__Heading___83D3K css-acwcvw f36-margin-bottom--m"
        data-test-id="cf-ui-heading"
      >
        About Image Focal Point
      </h1>
      <p
        class="Paragraph__Paragraph___2aO-9 f36-margin-bottom--m"
        data-test-id="cf-ui-paragraph"
      >
        The Image Focal Point app allows you to associate focal point data with uploaded image assets, to achieve better cropping amongst different devices and screen sizes.
      </p>
      <hr
        class="css-14380qe"
      />
    </div>
  </div>
  <div
    class="css-13jvbbd"
  >
    <img
      alt="logo"
      src="app-logo.svg"
    />
  </div>
</div>
`;

exports[`AppView when the app is installed should render the app view with configuration screen 1`] = `
<div>
  <div
    class="css-1cyywdw"
  />
  <div
    class="css-jwwzlh"
  >
    <div
      class="Typography__Typography___1ZUfE"
      data-test-id="cf-ui-text-container"
    >
      <h1
        class="Heading__Heading___83D3K css-acwcvw f36-margin-bottom--m"
        data-test-id="cf-ui-heading"
      >
        About Image Focal Point
      </h1>
      <p
        class="Paragraph__Paragraph___2aO-9 f36-margin-bottom--m"
        data-test-id="cf-ui-paragraph"
      >
        The Image Focal Point app allows you to associate focal point data with uploaded image assets, to achieve better cropping amongst different devices and screen sizes.
      </p>
      <hr
        class="css-14380qe"
      />
      <h1
        class="Heading__Heading___83D3K css-acwcvw f36-margin-bottom--m"
        data-test-id="cf-ui-heading"
      >
        Getting started
      </h1>
      <p
        class="Paragraph__Paragraph___2aO-9 f36-margin-bottom--m"
        data-test-id="cf-ui-paragraph"
      >
        Follow the next steps to create a new image entry with focal point data:
      </p>
      <ul
        class="List__List___1awv4 css-qslnu8"
        data-test-id="cf-ui-list"
      >
        <li
          class="ListItem__ListItem___3S9Oc"
          data-test-id="cf-ui-list-item"
        >
          Go to the "Content" page
        </li>
        <li
          class="ListItem__ListItem___3S9Oc"
          data-test-id="cf-ui-list-item"
        >
          Create a new entry of type "Image with Focal Point" (or the name you chose during the installation)
        </li>
        <li
          class="ListItem__ListItem___3S9Oc"
          data-test-id="cf-ui-list-item"
        >
          Fill in the required fields and publish it
        </li>
      </ul>
      <p
        class="Paragraph__Paragraph___2aO-9 css-15g2oxy f36-margin-bottom--m"
        data-test-id="cf-ui-paragraph"
      >
        Follow the next steps to reference the newly created image with focal point entry in another content type:
      </p>
      <ul
        class="List__List___1awv4 css-qslnu8"
        data-test-id="cf-ui-list"
      >
        <li
          class="ListItem__ListItem___3S9Oc"
          data-test-id="cf-ui-list-item"
        >
          Go to the "Content model" page
        </li>
        <li
          class="ListItem__ListItem___3S9Oc"
          data-test-id="cf-ui-list-item"
        >
          Edit the content type that needs to reference the image with the focal point
        </li>
        <li
          class="ListItem__ListItem___3S9Oc"
          data-test-id="cf-ui-list-item"
        >
          Create a new field of type "Reference"
        </li>
        <li
          class="ListItem__ListItem___3S9Oc"
          data-test-id="cf-ui-list-item"
        >
          Set a validation rule requiring that the content type the reference points to is of type "Image with Focal Point"
        </li>
        <li
          class="ListItem__ListItem___3S9Oc"
          data-test-id="cf-ui-list-item"
        >
          Optionally set the appearance of the new reference field to "Entry Card" in its settings
        </li>
        <li
          class="ListItem__ListItem___3S9Oc"
          data-test-id="cf-ui-list-item"
        >
          Go back to the "Content" page
        </li>
        <li
          class="ListItem__ListItem___3S9Oc"
          data-test-id="cf-ui-list-item"
        >
          Create/edit an entry of the content type referencing the image with the focal point
        </li>
        <li
          class="ListItem__ListItem___3S9Oc"
          data-test-id="cf-ui-list-item"
        >
          Set the reference field you added to point to the image with focal point entry
        </li>
      </ul>
      <hr
        class="css-14380qe"
      />
      <h1
        class="Heading__Heading___83D3K css-acwcvw f36-margin-bottom--m"
        data-test-id="cf-ui-heading"
      >
        Uninstallation and cleanup
      </h1>
      <p
        class="Paragraph__Paragraph___2aO-9 f36-margin-bottom--m"
        data-test-id="cf-ui-paragraph"
      >
        If you uninstall the Image Focal Point app you will have to manually clean up the content type we created for you during the installation. To do that follow the next steps:
      </p>
      <ul
        class="List__List___1awv4 css-qslnu8"
        data-test-id="cf-ui-list"
      >
        <li
          class="ListItem__ListItem___3S9Oc"
          data-test-id="cf-ui-list-item"
        >
          Go to the "Content" page
        </li>
        <li
          class="ListItem__ListItem___3S9Oc"
          data-test-id="cf-ui-list-item"
        >
          Delete any entries of type "Image with Focal Point" (or the name you chose during installation)
        </li>
        <li
          class="ListItem__ListItem___3S9Oc"
          data-test-id="cf-ui-list-item"
        >
          Go to the "Content model" page
        </li>
        <li
          class="ListItem__ListItem___3S9Oc"
          data-test-id="cf-ui-list-item"
        >
          Edit the content type the app created, select "Actions" from the top menu and "Delete"
        </li>
      </ul>
    </div>
  </div>
  <div
    class="css-13jvbbd"
  >
    <img
      alt="logo"
      src="app-logo.svg"
    />
  </div>
</div>
`;

exports[`AppView when the app is not installed should render inline validation if the content type id is taken 1`] = `
<div
  class="TextField__TextField___Sf6eo css-15g2oxy"
  data-test-id="content-type-name"
>
  <div
    class="TextField__TextField__label-wrapper___2-MJT"
  >
    <label
      class="FormLabel__FormLabel___3d6zQ"
      data-test-id="cf-ui-form-label"
      for="content-type-name"
    >
      Content type name
      <span
        class="FormLabel__FormLabel__required-text___3mvdm"
      >
        (
        required
        )
      </span>
    </label>
  </div>
  <div
    class="TextInput__TextInput___36-K- TextInput__TextInput--full___1EJEW TextInput__TextInput--negative___iVq__"
  >
    <input
      aria-label="contentTypeName"
      class="TextInput__TextInput__input___27vDB a11y__focus-border--default___60AXp"
      data-test-id="content-type-name-input"
      id="content-type-name"
      name="contentTypeName"
      placeholder="e.g. Image with Focal Point"
      required=""
      value="Image with Focal Point"
    />
  </div>
  <div
    class="ValidationMessage__ValidationMessage___3_rEq TextField__TextField__validation-message___1Idkl"
    data-test-id="cf-ui-validation-message"
  >
    <svg
      class="Icon__Icon___38Epv ValidationMessage__ValidationMessage__icon___3HPCh Icon__Icon--small___1yGZK Icon__Icon--negative___1dled"
      data-test-id="cf-ui-icon"
      height="18"
      viewBox="0 0 24 24"
      width="18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
      />
    </svg>
    <p
      class="ValidationMessage__ValidationMessage__text___8FBj5"
    >
      A content type with ID "imageWithFocalPoint" already exists. Try a different name.
    </p>
  </div>
  <div
    class="TextField__TextField__hints___3Di2P"
  >
    <p
      class="HelpText__HelpText___uWbja TextField__TextField__help-text___p4rVf"
      data-test-id="cf-ui-help-text"
    >
      You can use this content type to wrap images with focal point data
    </p>
  </div>
</div>
`;

exports[`AppView when the app is not installed should render inline validation if the content type id is taken 2`] = `
<div
  class="TextField__TextField___Sf6eo css-15g2oxy"
  data-test-id="content-type-id"
>
  <div
    class="TextField__TextField__label-wrapper___2-MJT"
  >
    <label
      class="FormLabel__FormLabel___3d6zQ"
      data-test-id="cf-ui-form-label"
      for="content-type-id"
    >
      Content type ID
      <span
        class="FormLabel__FormLabel__required-text___3mvdm"
      >
        (
        required
        )
      </span>
    </label>
  </div>
  <div
    class="TextInput__TextInput___36-K- TextInput__TextInput--full___1EJEW TextInput__TextInput--negative___iVq__"
  >
    <input
      aria-label="contentTypeId"
      class="TextInput__TextInput__input___27vDB a11y__focus-border--default___60AXp"
      data-test-id="content-type-id-input"
      id="content-type-id"
      name="contentTypeId"
      required=""
      value="imageWithFocalPoint"
    />
  </div>
  <div
    class="ValidationMessage__ValidationMessage___3_rEq TextField__TextField__validation-message___1Idkl"
    data-test-id="cf-ui-validation-message"
  >
    <svg
      class="Icon__Icon___38Epv ValidationMessage__ValidationMessage__icon___3HPCh Icon__Icon--small___1yGZK Icon__Icon--negative___1dled"
      data-test-id="cf-ui-icon"
      height="18"
      viewBox="0 0 24 24"
      width="18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
      />
    </svg>
    <p
      class="ValidationMessage__ValidationMessage__text___8FBj5"
    >
      A content type with ID "imageWithFocalPoint" already exists. Try a different ID.
    </p>
  </div>
  <div
    class="TextField__TextField__hints___3Di2P"
  >
    <p
      class="HelpText__HelpText___uWbja TextField__TextField__help-text___p4rVf"
      data-test-id="cf-ui-help-text"
    >
      The ID is generated from the name, you can also set it manually
    </p>
  </div>
</div>
`;

exports[`AppView when the app is not installed should render the app view with installation screen 1`] = `
<div>
  <div
    class="css-1cyywdw"
  />
  <div
    class="css-jwwzlh"
  >
    <div
      class="Typography__Typography___1ZUfE"
      data-test-id="cf-ui-text-container"
    >
      <h1
        class="Heading__Heading___83D3K css-acwcvw f36-margin-bottom--m"
        data-test-id="cf-ui-heading"
      >
        About Image Focal Point
      </h1>
      <p
        class="Paragraph__Paragraph___2aO-9 f36-margin-bottom--m"
        data-test-id="cf-ui-paragraph"
      >
        The Image Focal Point app allows you to associate focal point data with uploaded image assets, to achieve better cropping amongst different devices and screen sizes.
      </p>
      <hr
        class="css-14380qe"
      />
      <h1
        class="Heading__Heading___83D3K css-acwcvw f36-margin-bottom--m"
        data-test-id="cf-ui-heading"
      >
        Configuration
      </h1>
      <p
        class="Paragraph__Paragraph___2aO-9 f36-margin-bottom--m"
        data-test-id="cf-ui-paragraph"
      >
        To help you get started, we are going to create a content type for you with a title field, an image field and a focal point field.
      </p>
      <div
        class="TextField__TextField___Sf6eo css-15g2oxy"
        data-test-id="content-type-name"
      >
        <div
          class="TextField__TextField__label-wrapper___2-MJT"
        >
          <label
            class="FormLabel__FormLabel___3d6zQ"
            data-test-id="cf-ui-form-label"
            for="content-type-name"
          >
            Content type name
            <span
              class="FormLabel__FormLabel__required-text___3mvdm"
            >
              (
              required
              )
            </span>
          </label>
        </div>
        <div
          class="TextInput__TextInput___36-K- TextInput__TextInput--full___1EJEW"
        >
          <input
            aria-label="contentTypeName"
            class="TextInput__TextInput__input___27vDB a11y__focus-border--default___60AXp"
            data-test-id="content-type-name-input"
            id="content-type-name"
            name="contentTypeName"
            placeholder="e.g. Image with Focal Point"
            required=""
            value="Image with Focal Point"
          />
        </div>
        <div
          class="TextField__TextField__hints___3Di2P"
        >
          <p
            class="HelpText__HelpText___uWbja TextField__TextField__help-text___p4rVf"
            data-test-id="cf-ui-help-text"
          >
            You can use this content type to wrap images with focal point data
          </p>
        </div>
      </div>
      <div
        class="TextField__TextField___Sf6eo css-15g2oxy"
        data-test-id="content-type-id"
      >
        <div
          class="TextField__TextField__label-wrapper___2-MJT"
        >
          <label
            class="FormLabel__FormLabel___3d6zQ"
            data-test-id="cf-ui-form-label"
            for="content-type-id"
          >
            Content type ID
            <span
              class="FormLabel__FormLabel__required-text___3mvdm"
            >
              (
              required
              )
            </span>
          </label>
        </div>
        <div
          class="TextInput__TextInput___36-K- TextInput__TextInput--full___1EJEW"
        >
          <input
            aria-label="contentTypeId"
            class="TextInput__TextInput__input___27vDB a11y__focus-border--default___60AXp"
            data-test-id="content-type-id-input"
            id="content-type-id"
            name="contentTypeId"
            required=""
            value="imageWithFocalPoint"
          />
        </div>
        <div
          class="TextField__TextField__hints___3Di2P"
        >
          <p
            class="HelpText__HelpText___uWbja TextField__TextField__help-text___p4rVf"
            data-test-id="cf-ui-help-text"
          >
            The ID is generated from the name, you can also set it manually
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="css-13jvbbd"
  >
    <img
      alt="logo"
      src="app-logo.svg"
    />
  </div>
</div>
`;
